### PR TITLE
fix: remove optional for SearchHit Vector

### PR DESCRIPTION
**Pull request by honeyAcorn**
We remove the whole flag and create a new API.

### CODE REVIEW

**Review by malandis — APPROVED** [review]

**Comment by honeyAcorn** [comment]
Having that `null` in the return result is ugly, I'm working on changing it to oneof to fix it. So will not merge this one yet.

**Comment by honeyAcorn** [comment]
Change it to oneof type so it doesn't just print null. https://github.com/momentohq/client-protos/pull/229/commits/218169b137e204734b57881d575935f16eb2ffe5

**Comment by honeyAcorn** [comment]
Based on decision here: https://gomomento.slack.com/archives/C05HC8Y7FS8/p1698175193146069
We decide to come up with second search API. Please help me review it with the name @malandis.

**Comment by honeyAcorn** [comment]
@malandis @cprice404 I choose the SearchFetchVectors (over SearchRetreiveVectors) for now because you mentioned it's for the whole collection, I think that applies because we are fetching the vectors for all the search result data. With that being said, feel free to suggest other names too before I merge it!

**Review by cprice404 — DISMISSED** [review]

File: proto/vectorindex.proto
Comment (malandis) [inline proto/vectorindex.proto:9]: Should the rpc be called `SearchAndFetchVector**s**`?

Comment (honeyAcorn) [reply proto/vectorindex.proto]: Sharp eyes! Yes! https://github.com/momentohq/client-protos/pull/229/commits/d0d4271d68423d42fbb467981e8884b1523b66d2

**Review by malandis — APPROVED** [review]
LGTM 👍

**Pull request merged**
